agent/verify_job.go

```diff
@@ -105,7 +105,7 @@ func (r *JobRunner) verifyJob(keySet jwk.Set) error {
 			// jcs.Transform chokes on "" and "null", and json.Marshal encodes
 			// nil slice as "null", but zero-length slice as "[]".
 			emptyStepPlugins := len(step.Plugins) == 0
-			emptyJobPlugins := (jobPluginsJSON == "" || jobPluginsJSON == "null" || jobPluginsJSON == "[]")
```

Comment:
Hmm... I don't know if this can be removed, even with the updated JCS lib. If `jobPluginsJSON == "null"`, but `step.Plugins` was an empty slice, then `emptyStepPlugins` would be true but `emptyJobPlugins` would be false, and then after marshalling and canonicalisation, would compare `null` against `[]`.

But maybe it doesn't matter? If there are no plugins then the backend should not even give us a `BUILDKITE_PLUGINS` env var.